lib/Graphs/SVFG.cpp

```diff
@@ -252,38 +256,49 @@ void SVFG::addSVFGNodesForAddrTakenVars()
     for(MemSSA::BBToPhiSetMap::iterator it = mssa->getBBToPhiSetMap().begin(),
             eit = mssa->getBBToPhiSetMap().end(); it!=eit; ++it)
     {
-        for(PHISet::iterator pi = it->second.begin(), epi = it->second.end(); pi!=epi; ++pi)
-            addIntraMSSAPHISVFGNode(*pi);
+        for(PHISet::iterator pi = it->second.begin(), epi = it->second.end(); pi!=epi; ++pi){
+            MemSSA::PHI* phi =  *pi;
+            addIntraMSSAPHISVFGNode(phi->getBasicBlock(), phi->opVerBegin(), phi->opVerEnd(),phi->getResVer());
+        }
     }
     /// initialize memory SSA entry chi nodes
     for(MemSSA::FunToEntryChiSetMap::iterator it = mssa->getFunToEntryChiSetMap().begin(),
             eit = mssa->getFunToEntryChiSetMap().end(); it!=eit; ++it)
     {
-        for(CHISet::iterator pi = it->second.begin(), epi = it->second.end(); pi!=epi; ++pi)
-            addFormalINSVFGNode(SVFUtil::cast<ENTRYCHI>(*pi));
+        for(CHISet::iterator pi = it->second.begin(), epi = it->second.end(); pi!=epi; ++pi){
+            const MemSSA::ENTRYCHI* chi = SVFUtil::cast<ENTRYCHI>(*pi);
+            addFormalINSVFGNode(new FunEntryBlockNode(totalSVFGNode++,chi->getFunction()), chi->getResVer());
```

Comment:
Do not create new FunEntryBlockNode. Use `pag->getICFG()->getFunEntryBlockNode(chi->getFunction())` instead.